ports/atmel-samd/boards/sam32/mpconfigboard.h

```diff
@@ -1,23 +1,22 @@
-
-#define MICROPY_HW_BOARD_NAME "SAM32v2a"
+#define MICROPY_HW_BOARD_NAME "SAM32v26"
 #define MICROPY_HW_MCU_NAME "samd51j20"
 #define CIRCUITPY_MCU_FAMILY samd51
 
 #define MICROPY_HW_LED_STATUS   (&pin_PA27)
-#define MICROPY_HW_NEOPIXEL 	(&pin_PA15)
+#define MICROPY_HW_NEOPIXEL     (&pin_PA15)
 
 #define MICROPY_PORT_A        (PORT_PA24 | PORT_PA25)
 #define MICROPY_PORT_B        (0)
 #define MICROPY_PORT_C        (0)
-#define MICROPY_PORT_D 	      (0)
+#define MICROPY_PORT_D        (0)
 
-// No microcontroller.nvm
-#define CIRCUITPY_INTERNAL_NVM_SIZE 0
+#define CIRCUITPY_INTERNAL_NVM_SIZE 1024
```

Comment:
`CIRCUITPY_INTERNAL_NVM_SIZE` should be a multiple of 8192 on the SAMD51, because that's the page erase size. When bytes are written to the nvm, the page is erased and then the undisturbed data is written back, so if there's vital data in the erased part, it will temporarily be all ones.

We should put a compile-time check in for this restriction in mpconfigport.h to catch this -- sorry!

Reply:
Corrected above: it's block size, not page size.

Reply:
I see! I had not idea this was a limitation. I'll fix and request a re-review. Thanks @dhalbert!

Reply:
I fixed the automatic check (it was using the wrong number) and didn't catch the problem; should work in the future: #2621.